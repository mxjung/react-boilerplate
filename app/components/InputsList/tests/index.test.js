import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from 'react-testing-library';

import InputsLIst from '../index';
import configureStore from '../../../configureStore';

describe('<InputsLIst />', () => {
  it('should render the loading indicator when its loading', () => {
    const { container } = render(<InputsLIst loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <InputsLIst loading={false} error={{ message: 'Loading failed!' }} />
      </IntlProvider>,
    );
    expect(queryByText(/Something went wrong/)).not.toBeNull();
  });

  it('should render the inputs if loading was successful', () => {
    const store = configureStore({ global: null }, browserHistory);
    const inputs = ['test'];
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <InputsLIst inputs={inputs} error={false} />
        </IntlProvider>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render anything if nothing interesting is provided', () => {
    const { container } = render(
      <InputsLIst inputs={false} error={false} loading={false} />,
    );

    expect(container.firstChild).toBeNull();
  });
});
