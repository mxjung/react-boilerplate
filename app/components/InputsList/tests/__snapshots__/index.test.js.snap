// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputsLIst /> should render the inputs if loading was successful 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 laCwDT"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <li
      class="Wrapper-euo0oy-0 jLvxtc"
    >
      <div
        class="Item-sc-3y9mie-0 fKXSkT"
      >
        test
      </div>
    </li>
  </ul>
</div>
`;

exports[`<InputsLIst /> should render the loading indicator when its loading 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 laCwDT"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <div
      class="Wrapper-sc-12uw37d-0 bTWIVF"
    >
      <div
        class="Circle__CirclePrimitive-ww56dy-0 enQIMS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jCxmwS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 dyFXB"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ikwUeP"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hwlQRz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jjXxyb"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jVExkJ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 haCVIz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ghhiLZ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 bCbqdt"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hZPQMA"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 fsGOmP"
      />
    </div>
  </ul>
</div>
`;
